NEWS.md

```diff
@@ -141,6 +141,27 @@
 
 26. `base::droplevels()` gains a fast method for `data.table`, [#647](https://github.com/Rdatatable/data.table/issues/647). Thanks to Steve Lianoglou for requesting, and Jan Gorecki and Benjamin Schwendinger for the PR. `fdroplevels()` for use on vectors has also been added.
 
+27. `first` and `last` gain `na.rm` argument to say whether `NA` should be ignored for finding first/last, [#4446](https://github.com/Rdatatable/data.table/issues/4446). The additional argument `na.rm` is also added to the GForce optimized versions of `first` and `last` where `NA` is still returned as first/last for groups only consisting of `NA`s, [#4239](https://github.com/Rdatatable/data.table/issues/4239). Thanks to Nicolas Bennett and Michael Chirico for the request, and Benjamin Schwendinger for the PR.
+
+    ```R
+    DT = data.table(a=c(1:3,NA), b=1:2, c=c(1L, rep(NA, 3)))
+    DT
+    #       a     b     c
+    #   <int> <int> <int>
+    #1:     1     1     1
+    #2:     2     2    NA
+    #3:     3     1    NA
+    #4:    NA     2    NA
+    last(DT, na.rm=TRUE)
+    #       a     b     c
+    #   <int> <int> <int>
+    #1:     1     1     1
+    DT[, last(.SD, na.rm=TRUE), b]
+    #       b     a     c
+    #   <int> <int> <int>
+    #1:     1     3     1
+    #2:     2     2    NA
```

Comment:
On this last output I was expecting : 
```
#        b     a     c
#    <int> <int> <int>
# 1:     1     1     1
# 2:     2     NA   NA
```
because I was expecting it to return the last row which has no NA in any column (for the b==1 group there is one row like, row 1).  For the b==2 group, there are no rows in .SD which have no NA, so I expected the last row which is row 4.  Am I misunderstanding something?

The example could show a simple vector first as a one or two line example, before moving on to the multi-column example.

Reply:
At first I was expecting the same, but since `last(.SD, na.rm=TRUE)` is getting lapply optimised to `list(last(a, na.rm=TRUE), last(c, na.rm=TRUE))` it explains the current behavior. We could ofc turn off lapply optimization in the case of `.SD` and `na.rm=TRUE` but this would also reduce the usage for `glast`.

I guess its unambigous to expect:
```R
DT = data.table(a=c(1:3,NA), b=1:2, c=c(1L, rep(NA, 3)))
DT
#        a     b     c
#    <int> <int> <int>
# 1:     1     1     1
# 2:     2     2    NA
# 3:     3     1    NA
# 4:    NA     2    NA
DT[, last(a, na.rm=TRUE), b]
#        b    V1
#    <int> <int>
# 1:     1     3
# 2:     2     2
DT[, last(c, na.rm=TRUE), b]
#        b    V1
#    <int> <int>
# 1:     1     1
# 2:     2    NA
```